packages/flutter/test/widgets/nested_scroll_view_test.dart

```diff
@@ -644,39 +743,467 @@ void main() {
+      final double scrollExtent = appBarHeight + 50.0;
+      expect(globalKey.currentState.outerController.offset, 0.0);
+      expect(globalKey.currentState.innerController.offset, 0.0);
+
+      // The scroll gesture should occur in the inner body, so the whole
+      // scroll view is scrolled.
+      final TestGesture gesture = await tester.startGesture(Offset(
+        0.0,
+        appBarHeight + 1.0,
+      ));
+      await gesture.moveBy(Offset(0.0, -scrollExtent));
+      await tester.pump();
+
+      appBarHeight = tester.renderObject<RenderBox>(find.byType(AppBar)).size.height;
+      // This is an expanding AppBar.
+      expect(appBarHeight, 104.0);
+      // The outer scroll controller should show an offset of the applied
+      // scrollExtent.
+      expect(globalKey.currentState.outerController.offset, 200.0);
+      // the inner scroll controller should have scrolled equivalent to the
+      // difference between the applied scrollExtent and the outer extent.
+      expect(globalKey.currentState.innerController.offset, 50.0);
+    });
+
+    testWidgets('NestedScrollViewState.outerController should correspond to NestedScrollView.controller', (
+      WidgetTester tester) async {
+      final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+      final ScrollController scrollController = ScrollController();
+
+      await tester.pumpWidget(buildTest(
+        controller: scrollController,
+        key: globalKey,
+      ));
+
+      // Scroll to compare offsets between controllers.
+      final TestGesture gesture = await tester.startGesture(const Offset(
+        0.0,
+        100.0,
+      ));
+      await gesture.moveBy(const Offset(0.0, -100.0));
+      await tester.pump();
+
+      expect(
+        scrollController.offset,
+        globalKey.currentState.outerController.offset,
+      );
+      expect(
+        tester.widget<NestedScrollView>(find.byType(NestedScrollView)).controller.offset,
+        globalKey.currentState.outerController.offset,
+      );
+    });
+
+    group('manipulating controllers when', () {
+      testWidgets('outer: not scrolled, inner: not scrolled', (WidgetTester tester) async {
+        final GlobalKey<NestedScrollViewState> globalKey1 = GlobalKey();
+        await tester.pumpWidget(buildTest(
+          key: globalKey1,
+          expanded: false,
+        ));
+        expect(globalKey1.currentState.outerController.position.pixels, 0.0);
+        expect(globalKey1.currentState.innerController.position.pixels, 0.0);
+        final double appBarHeight = tester.renderObject<RenderBox>(find.byType(AppBar)).size.height;
+
+        // Manipulating Inner
+        globalKey1.currentState.innerController.jumpTo(100.0);
+        expect(globalKey1.currentState.innerController.position.pixels, 100.0);
+        expect(
+          globalKey1.currentState.outerController.position.pixels,
+          appBarHeight,
+        );
+        globalKey1.currentState.innerController.jumpTo(0.0);
```

Comment:
Looks like the documented behavior is true.

Reply:
Do you think this should have more documentation to make it more understandable?